File: raster-go/model3d/scene.go
package model3d

import "raster-go/gl"

type Scene struct {
	Screen  *Screen
	ZBuffer *gl.ZBuffer
	Camera  *Camera
	Child   []*Node
}

func NewScene(width, height int) *Scene {
	camera := NewCamera()
	camera.SetPosition(0, 100, 0)

	zbuff := gl.NewZBuffer(width, height, 1)

	screen := NewScreen(width, height)

	return &Scene{Camera: camera, ZBuffer: zbuff, Screen: screen, Child: make([]*Node, 0, 0)}
}

func (s *Scene) AddChild(node *Node) {
	s.Child = append(s.Child, node)
}

File: raster-ts/src/scene/scene.ts
import { Camera } from "./camera";
import { Node } from "./node";

export class Scene {
    private camera: Camera;
    private child: Array<Node> = [];

    constructor() {
        this.camera = new Camera();
    }

    public addChild(node: Node): void {
        this.child.push(node);
    }

    public getChild(index: number): Node {
        return this.child[index];
    }

    public size(): number {
        return this.child.length;
    }

    public getCamera(): Camera {
        return this.camera;
    }
}